### DISCUSSION TITLE
What is HTTPClientError.deadlineExceeded when uploading?

### DISCUSSION THREAD

**Discussion by JetForMe** [soto-project/soto#719] [Q&A] [answered]
This little app I'm working on generally works well, but I've only ever used it from home, where the internet is okay. I'm now using it from another location where the internet upload is like 5 MB, and I'm getting `HTTPClientError.deadlineExceeded` when I try to upload a 34 MB file. Very small files upload just fine. Is this a timeout? Why would it do that if it’s getting bytes, albeit slowly? This is going to DigitalOcean Spaces, so please let me know if it's something I need to take up with them.

**Comment by adam-fowler** [comment, marked as answer]
Yes that'll be a timeout. You can get around that by setting the timeout when initializing the AWS service.
```swift
let s3 = S3(
    client: client,
    timeout: .minutes(5)
)
```

**Reply by JetForMe** [reply]
Ah! It's in the client setup. I was trying to find a timeout in the request setup. Thanks!

**Reply by adam-fowler** [reply]
For a temporary timeout you can use
```swift
s3.with(timeout: .minutes(5)).putObject(...)
```
